libbpf-tools/ksnoop.bpf.c

```diff
@@ -313,7 +314,13 @@ static int ksnoop(struct pt_regs *ctx, bool entry)
 		return 0;
 	trace->pid = currpid;
 
-	trace->cpu = bpf_get_smp_processor_id();
+        task = (struct task_struct *)bpf_get_current_task();
+        trace->pid_ns_id =
+            (__u32)BPF_CORE_READ(task, nsproxy, pid_ns_for_children, ns.inum);
```

Comment:
https://github.com/torvalds/linux/blob/v5.14/include/linux/nsproxy.h#L19-L21

According to the comments, this may not be right.

Reply:
Thanks @chenhengqi !
Will follow task_active_pid_ns()'s way to get pid namespace.

Reply:
updated